### PR TITLE
#169698715 user details

### PR DESCRIPTION
**Pull request by urbanishimwe**
#### What does this PR do?
- Get details of a user
#### Description of Task to be completed?
As a Visitor, I should Be able to get details about a certain User in BareFoot Nomad
#### How should this be manually tested?
routes:
**GET:**`/api/v1/users/profile/:id`
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[story](https://www.pivotaltracker.com/story/show/169698715)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: src/middlewares/userMiddlware.js [hunk 1]
```diff
@@ -41,6 +41,19 @@ const User = {
...
   },
 
+  async getUserById(req, res, next) {
+    const { id } = req.params;
+    const user = await db.Users.findOne({
+      where: { id },
+      raw: true,
+    });
+
+    if (!user) return sendResult(res, 409, 'User with id not found');
```
Comment (lemoissonM) [inline src/middlewares/userMiddlware.js]: change status to 404

**Review by lemoissonM — CHANGES_REQUESTED** [review]

File: src/middlewares/userMiddlware.js (hunk 1 above)
Comment (urbanishimwe) [reply src/middlewares/userMiddlware.js]: great

**Review by swaibat — APPROVED** [review]

File: src/middlewares/userMiddlware.js [hunk 2]
```diff
@@ -41,6 +41,19 @@ const User = {
     next();
   },
 
+  async getUserById(req, res, next) {
+    const { id } = req.params;
+    const user = await db.Users.findOne({
```
Comment (swaibat) [inline src/middlewares/userMiddlware.js]: @ great job urban but at least export this to `user.service.js` since its one of the `CRUD` operations

**Review by swaibat — CHANGES_REQUESTED** [review]

File: src/middlewares/userMiddlware.js (hunk 2 above)
Comment (urbanishimwe) [reply src/middlewares/userMiddlware.js]: great!

Comment (urbanishimwe) [reply src/middlewares/userMiddlware.js]: Luckily that service is already there

Comment (swaibat) [reply src/middlewares/userMiddlware.js]: cool

**Review by swaibat — APPROVED** [review]

**Review by lemoissonM — APPROVED** [review]

**Pull request merged**